### PR TITLE
Inline Documentation

### PR DESCRIPTION
**Pull request by dylanireland**
Inline, TypeDoc compliant documentation written for the following files

[CasperServiceByJsonRPC.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/inline-documentation/src/services/CasperServiceByJsonRPC.ts)
[CasperClient.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/inline-documentation/src/lib/CasperClient.ts)
[Contracts.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/inline-documentation/src/lib/Contracts.ts)
[DeployUtil.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/inline-documentation/src/lib/DeployUtil.ts)
[ByteConverters.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/inline-documentation/src/lib/ByteConverters.ts)

### CODE REVIEW

File: src/lib/CasperClient.ts [hunk 1]
```diff
@@ -9,13 +9,18 @@ import { BigNumber } from '@ethersproject/bignumber';
...
+   */
   constructor(nodeUrl: string) {
     this.nodeClient = new CasperServiceByJsonRPC(nodeUrl);
   }
 
   /**
-   * Generate new key pair.
-   * @param algo Currently we support Ed25519 and Secp256K1.
+   * Generate a new key pair
+   * @param algo The signature algorithm of the account. Currently we support Ed25519 and Secp256K1
```
Comment (hoffmannjan) [inline src/lib/CasperClient.ts]: maybe we can add the info that possible values are `SignatureAlgorithm.Ed25519` and `SignatureAlgorithm.Secp256K1`, what do you think?

File: src/lib/CasperClient.ts [hunk 2]
```diff
@@ -30,9 +35,9 @@ export class CasperClient {
 
   /**
    * Load private key from file
-   *
-   * @param path the path to the publicKey file
-   * @param algo the signature algorithm of the file
+   * @param path The path to the publicKey file
+   * @param algo The signature algorithm of the account. Currently we support Ed25519 and Secp256K1
+   * @returns New key pair with the specified SignatureAlgorithm
```
Comment (hoffmannjan) [inline src/lib/CasperClient.ts]: same as above

File: src/lib/CasperClient.ts [hunk 3]
```diff
@@ -114,11 +123,12 @@ export class CasperClient {
...
-   * Construct a unsigned Deploy object
-   *
-   * @param deployParams Parameters for deploy
-   * @param session
-   * @param payment
+   * Construct an unsigned Deploy object from the deploy parameters, session logic, and payment logic
+   * @param deployParams Deploy parameters
+   * @param session Session logic
+   * @param payment Payment logic
+   * @returns A deployable Deploy object
```
Comment (hoffmannjan) [inline src/lib/CasperClient.ts]: In fact its not `deployable` as it's not signed ;)

File: src/lib/CasperClient.ts [hunk 4]
```diff
@@ -140,34 +152,37 @@ export class CasperClient {
   /**
    * Send deploy to network
    * @param signedDeploy Signed deploy object
+   * @returns The sent Deploy's transaction hash, as a string
```
Comment (hoffmannjan) [inline src/lib/CasperClient.ts]: hex representation of transaction hash :)

File: src/lib/Contracts.ts [hunk 5]
```diff
@@ -10,25 +10,41 @@ import { DEFAULT_DEPLOY_TTL } from '../constants';
...
+   * Constructor
+   * @param casperClient The `CasperClient` object connected to a live node
+   */
   constructor(public casperClient?: CasperClient) {}
 
+  /**
+   * Attaches an on-chain smart contract to this `Contract` object using its hexadecimal string typed hash. The contract hash must include the prefix "hash-"
+   * @param contractHash The hexadecimal smart contract hash, with the prefix "hash-"
+   * @param contractPackageHash The hexadecimal smart contract package hash, with the prefix "hash-"
+   */
```
Comment (hoffmannjan) [inline src/lib/Contracts.ts:46]: you can add that `contractPackageHash` is an optional parameter and is only used when there is some event processing

File: src/lib/Contracts.ts [hunk 6]
```diff
@@ -46,6 +62,16 @@ export class Contract {
...
+  /**
+   * Install a smart contract on a Casper Network
+   * @param wasm `Uint8Array` representation of a WebAssembly compiled smart contract
+   * @param args The runtime arguments for the installment deploy
+   * @param paymentAmount The gas payment in motes, where 1 mote = 10^-9 CSPR. Use a stringified base-10 integer
+   * @param sender `CLPublicKey` of the sender of the installment deploy
+   * @param chainName The name of the network the installment deploy will be sent to. Use "casper" for the Casper mainnet, "casper-test" for the Casper official testnet, and "casper-net-*" for local NCTL instances, where * is equal to the id of the node you wish to deploy to.
+   * @param signingKeys An array of keypairs used to sign the deploy. If you are signing with one key, use an array with only the one keypair. If instead you are utilizing multi-sig functionality, provide multiple keypair objects in the array.
+   * @returns The installment deploy, to be sent to a node.
+   */
```
Comment (hoffmannjan) [inline src/lib/Contracts.ts:74]: really good, in future this method will be alias to some different method `callModuleBytesEntrypoint` :)

File: src/lib/Contracts.ts [hunk 7]
```diff
@@ -70,6 +96,17 @@ export class Contract {
     throw Error('You need to setContract before running this method.');
   }
 
+  /**
+   * Call an entrypoint of a smart contract.
+   * @param entryPoint The entrypoint of a smart contract that you wish to call
```
Comment (hoffmannjan) [inline src/lib/Contracts.ts]: the name of an entrypoint in fact

File: src/lib/Contracts.ts [hunk 8]
```diff
@@ -70,6 +96,17 @@ export class Contract {
...
+   * @param args The runtime arguments for the deploy
+   * @param sender `CLPublicKey` of the sender of the deploy
+   * @param chainName The name of the network the installment deploy will be sent to. Use "casper" for the Casper mainnet, "casper-test" for the Casper official testnet, and "casper-net-*" for local NCTL instances, where * is equal to the id of the node you wish to deploy to.
```
Comment (hoffmannjan) [inline src/lib/Contracts.ts]: this is true (currently) but I'm not sure if this should be part of the inline docs, as when the network names will change someday we will need to remember about this place in docs :). maybe its better to say that you can always check the network name by calling REST endpoint of a node, eg `:8888/status` and get the network name :)

**Review by hoffmannjan — CHANGES_REQUESTED** [review]
Awesome job @dylanireland , only few small issues that needs to be addressed. Cheers!

File: src/lib/CasperClient.ts [hunk 9]
```diff
@@ -30,9 +35,9 @@ export class CasperClient {
 
   /**
    * Load private key from file
-   *
-   * @param path the path to the publicKey file
-   * @param algo the signature algorithm of the file
+   * @param path The path to the publicKey file
+   * @param algo The signature algorithm of the account. The possible values are `SignatureAlgorithm.Ed25519` and SignatureAlgorithm.Secp256K1
+   * @returns New key pair with the specified SignatureAlgorithm
```
Comment (dylanireland) [inline src/lib/CasperClient.ts:40]: Needs updating

**Review by hoffmannjan — APPROVED** [review]

**Pull request merged**